service/controller/resource/spark/create.go

```diff
@@ -0,0 +1,308 @@
+		} else if err != nil {
+			return microerror.Mask(err)
+		}
+
+		h := sha512.New()
+		dataHash = fmt.Sprintf("%x", h.Sum(ignitionBlob))
+	}
+
+	var dataSecret *corev1.Secret
+	{
+		if sparkCR.Status.DataSecretName != "" {
+			var s corev1.Secret
+			err = r.ctrlClient.Get(ctx, client.ObjectKey{Namespace: cr.Namespace, Name: sparkCR.Status.DataSecretName}, &s)
+			if errors.IsNotFound(err) {
+				// This is ok. We'll create it then.
+			} else if err != nil {
+				return microerror.Mask(err)
+			} else {
+				dataSecret = &s
+			}
+		}
+
+		// Create Secret for ignition data.
+		if dataSecret == nil {
+			dataSecret = &corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      secretName(sparkCR.Name),
+					Namespace: sparkCR.Namespace,
+				},
+				Data: map[string][]byte{
+					ignitionBlobKey: ignitionBlob,
+				},
+			}
+
+			err = r.ctrlClient.Create(ctx, dataSecret)
+			if err != nil {
+				return microerror.Mask(err)
+			}
+		}
+	}
+
+	{
+		var sparkStatusUpdateNeeded bool
+
+		if !sparkCR.Status.Ready {
+			sparkCR.Status.Ready = true
+			sparkStatusUpdateNeeded = true
+		}
+
+		if sparkCR.Status.DataSecretName != dataSecret.Name {
+			sparkCR.Status.DataSecretName = dataSecret.Name
+			sparkStatusUpdateNeeded = true
+		}
+
+		if sparkCR.Status.Verification.Hash != dataHash {
+			sparkCR.Status.Verification.Hash = dataHash
+			sparkCR.Status.Verification.Algorithm = "sha512"
+			sparkStatusUpdateNeeded = true
+		}
+
+		if sparkStatusUpdateNeeded {
+			err = r.ctrlClient.Status().Update(ctx, &sparkCR)
+			if err != nil {
+				return microerror.Mask(err)
+			}
+		}
+	}
+
+	{
+		if bytes.Compare(ignitionBlob, dataSecret.Data[ignitionBlobKey]) != 0 {
+			dataSecret.Data[ignitionBlobKey] = ignitionBlob
+
+			err = r.ctrlClient.Update(ctx, dataSecret)
+			if err != nil {
+				return microerror.Mask(err)
+			}
+		}
+	}
+
+	return nil
+}
+
+func (r *Resource) createIgnitionBlob(ctx context.Context, cr *expcapzv1alpha3.AzureMachinePool, sparkCR *corev1alpha1.Spark) ([]byte, error) {
+	cluster, err := r.getOwnerCluster(ctx, cr.ObjectMeta)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	azureCluster, err := r.getAzureCluster(ctx, cluster)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	machinePool, err := r.getOwnerMachinePool(ctx, cr.ObjectMeta)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	release, err := r.getRelease(ctx, machinePool.ObjectMeta)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	var clusterCerts certs.Cluster
+	{
+		clusterCerts, err = r.certsSearcher.SearchCluster(key.ClusterID(cluster))
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+	}
+
+	var cloudConfig cloudconfig.Interface
+	{
+		// TODO: Construct cloudconfig instance.
+		//
+		// It probably requires that some configuration settings are wired from
+		// configmap to resource. Some can be hopefully hardcoded here. Some
+		// should be pulled from the outside configmap later (such as OIDC).
```

Comment:
This is the tricky part. Theoretically it's nothing more than just wiring up big pile of configuration and pulling other from available CRs. OIDC config doesn't exist right now, but we need to find a place to put it. Then it can be also configured here.